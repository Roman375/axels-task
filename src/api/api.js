import axios from 'axios'

export const employeeAPI = {
  getEmployees() {
    return axios.get(`http://api.additivasia.io/api/v1/assignment/employees`)
  },
}

export const overviewAPI = {
  getOverviews(employeeName) {
    return axios.get(`http://api.additivasia.io/api/v1/assignment/employees/${employeeName}`)
  },
}
