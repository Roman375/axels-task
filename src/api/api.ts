import axios from 'axios'
import { fetchEmployeeType } from '../types/employeesType'
import { getOverviewsType } from '../types/overviewsType'

export const employeeAPI = {
  getEmployees(): Promise<fetchEmployeeType> {
    return axios.get(`http://api.additivasia.io/api/v1/assignment/employees`)
  },
}

export const overviewAPI = {
  getOverviews(employeeName: string): Promise<getOverviewsType> {
    return axios.get(`http://api.additivasia.io/api/v1/assignment/employees/${employeeName}`)
  },
}
