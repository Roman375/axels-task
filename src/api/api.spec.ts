import { employeeAPI, overviewAPI } from './api'

//Snapshot
describe('With Snapshot Testing', () => {
  it('get employeeAPI correctly', () => {
    const actual = employeeAPI.getEmployees()
    expect(actual).toMatchSnapshot()
  })

  it('get overviewAPI correctly', () => {
    const name = 'test'
    const actual = overviewAPI.getOverviews(name)
    expect(actual).toMatchSnapshot()
  })
})
