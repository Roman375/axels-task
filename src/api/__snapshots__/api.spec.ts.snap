// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing get employeeAPI correctly 1`] = `Promise {}`;

exports[`With Snapshot Testing get overviewAPI correctly 1`] = `Promise {}`;
